### PR TITLE
docs: major overhaul of contribution guidelines and documentation

### PR DESCRIPTION
**Pull request by dlaehnemann**
This pull request is a major overhaul of the contribution guidelines. It updates it to the current best practices and also cleans up the `I confirm that:` list in the pull request template. Hopefully, this makes contributing (and improving) wrappers a little bit easier.

### QC

* [x] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* the `environment.yaml` pinning has been updated by running `snakedeploy pin-conda-envs environment.yaml` on a linux machine,
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

**Comment by dlaehnemann** [comment]
@fgvieira, as you have been contributing a lot on the wrappers, would you be willing to have a look at this pull request? I'm trying to clean up the wrapper contribution docs here, to make them more comprehensive and accessible. At the same time, I want to move most things away from the pull request template to the docs, with only some important reminders in the pull request. In any case, it would be great if you could have a look, as I would think you are best placed to find gaps and mistakes!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -6,22 +6,12 @@
 ### QC
 <!-- Make sure that you can tick the boxes below. -->
 
-* [ ] I confirm that:
+* [ ] I confirm that I have followed the [documentation for contributing wrappers](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).
```
Comment (fgvieira) [inline .github/pull_request_template.md:9]:
```suggestion
* [ ] I confirm that I have followed the [documentation for contributing to snakemake-wrappers](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -6,22 +6,12 @@
...
-* `Snakefile`s pass the linting (`snakemake --lint`),
-* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
-* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
-* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).
+While the contributions guidelines are more extensive, please particularly ensure that:
+* [ ] `test.py` was updated to call any added or updated example rules in a `Snakefile`
+* [ ] `input:` and `output:` file paths in the rules can be chosen arbitrarily
+* [ ] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
+* [ ] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to 
+* [ ] the `meta.yaml` contains a link to the documentation of the respective tool or command,
```
Comment (fgvieira) [inline .github/pull_request_template.md:16]:
```suggestion
* [ ] the `meta.yaml` contains a link to the documentation of the respective tool or command (under `url:`)
```

File: docs/contributing.rst [hunk 3]
```diff
@@ -14,31 +14,59 @@ If you want to contribute we suggest the following procedure:
...
-  test in ``test.py``
-* ensure consistent `formatting`_ of Python files and `linting`_ of Snakefiles.
+
+In general, always take inspiration from existing wrappers.
+And then, contributions should:
+
+* provide the following files:
+
+  * ``meta.yaml`` (wrapper description), see :ref:`meta`
+  * ``environment.yaml`` (required software), see :ref:`environment`
```
Comment (fgvieira) [inline docs/contributing.rst:24]: Should we also refer here the environment pin file?

**Comment by fgvieira** [comment]
@dlaehnemann it looks quite good. Just made some comments/suggestions.

File: docs/contributing.rst (hunk 3 above)
Comment (dlaehnemann) [reply docs/contributing.rst:24]: Good point, will add this in a sec.

Comment (dlaehnemann) [reply docs/contributing.rst:24]: Done now, in https://github.com/snakemake/snakemake-wrappers/pull/3076/commits/29a40928de91e75be98932979840493822bc3334

**Comment by dlaehnemann** [comment]
@fgvieira Thanks for the review, I applied your suggestions (and found some more formatting things). Would you care for a final review?

**Review by fgvieira — APPROVED** [review]

**Comment by fgvieira** [comment]
@dlaehnemann looks good! Thanks for updating the docs.

**Pull request merged**